Sources/Common/Communication/Event.swift

```diff
@@ -24,71 +24,120 @@ public extension EventRepresentable {
+    public init(storageId: String = UUID().uuidString, identifier: String, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.identifier = identifier
         self.params = params
     }
 }
 
 public struct ScreenViewedEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
     public let name: String
 
-    public init(name: String, params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, name: String, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.name = name
         self.params = params
     }
 }
 
 public struct ResetEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
 
-    public init(params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.params = params
     }
 }
 
 public struct TrackMetricEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
     public let deliveryID: String
     public let event: String
     public let deviceToken: String
 
-    public init(deliveryID: String, event: String, deviceToken: String, params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, deliveryID: String, event: String, deviceToken: String, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.params = params
         self.deliveryID = deliveryID
         self.event = event
         self.deviceToken = deviceToken
     }
 }
 
+public struct TrackInAppMetricEvent: EventRepresentable {
+    public var storageId: String
+    public var params: [String: String] = [:]
+    public let deliveryID: String
+    public let event: String
+
+    public init(storageId: String = UUID().uuidString, deliveryID: String, event: String, params: [String: String] = [:]) {
+        self.storageId = storageId
+        self.params = params
+        self.deliveryID = deliveryID
+        self.event = event
+    }
+}
+
 public struct RegisterDeviceTokenEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
     public let token: String
 
-    public init(token: String, params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, token: String, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.token = token
         self.params = params
     }
 }
 
 public struct DeleteDeviceTokenEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
 
-    public init(params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.params = params
```

Comment:
Since `EventRepresentable ` implements `Codable` we need to have the constructor if we use `let` instead of `var`

From what I have read, since i am new to iOS and swift 😄 

When you declare a property in a Codable conforming struct with both let (making it immutable) and a default value, Swift's Decodable implementation won't be able to overwrite that default value during the decoding process. This is because immutable properties (let) can only be set once, and in your case, they are already set with a default value.